plugin/trino-http-query-listener/src/main/java/io/trino/plugin/httpquery/HttpQueryListenerConfig.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.httpquery;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.validation.FileExists;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpQueryListenerConfig
+{
+    private int retryCount = 2;
+    private Duration connectTimeout = Duration.ofSeconds(10);
+    private Duration readTimeout = Duration.ofSeconds(10);
+    private Duration writeTimeout = Duration.ofSeconds(10);
+
+    private final EnumSet<HttpQueryListenerEventType> loggedEvents = EnumSet.noneOf(HttpQueryListenerEventType.class);
+    private URL ingestUrl;
```

Comment:
Let's keep it String in config and parse on use.

Reply:
I'd rather have it like this so it can fail faster when a bad URL is given